event-handlers/src/main/java/no/sikt/nva/nvi/events/UpsertNviCandidateHandler.java

```diff
@@ -60,8 +67,44 @@ private static void validateRequiredCandidateDetails(CandidateDetails candidateD
         Objects.requireNonNull(candidateDetails.verifiedCreators());
     }
 
+    private static DbCandidate toPendingCandidate(CandidateEvaluatedMessage message) {
+        return DbCandidate.builder()
+                   .publicationBucketUri(message.publicationBucketUri())
+                   .publicationId(message.candidateDetails().publicationId())
+                   .applicable(true)
+                   .creators(mapToVerifiedCreators(message.candidateDetails().verifiedCreators()))
+                   .level(DbLevel.parse(message.candidateDetails().level()))
+                   .instanceType(message.candidateDetails().instanceType())
+                   .publicationDate(mapToPublicationDate(message.candidateDetails()
+                                                             .publicationDate()))
+                   .points(mapToInstitutionPoints(message.institutionPoints()))
+                   .build();
+    }
+
+    private static List<DbInstitutionPoints> mapToInstitutionPoints(Map<URI, BigDecimal> institutionPoints) {
+        return institutionPoints.entrySet().stream()
+                   .map(entry -> new DbInstitutionPoints(entry.getKey(), entry.getValue())).toList();
+    }
+
+    private static DbPublicationDate mapToPublicationDate(CandidateDetails.PublicationDate publicationDate) {
+        return new DbPublicationDate(publicationDate.year(), publicationDate.month(), publicationDate.day());
+    }
+
+    private static List<DbCreator> mapToVerifiedCreators(List<CandidateDetails.Creator> creators) {
+        return creators.stream()
+                   .map(
+                       verifiedCreatorDto -> new DbCreator(verifiedCreatorDto.id(),
+                                                           verifiedCreatorDto.nviInstitutions()))
+                   .toList();
+    }
+
     private void upsertNviCandidate(CandidateEvaluatedMessage evaluatedCandidate) {
-        nviService.upsertCandidate(evaluatedCandidate);
+        nviService.upsertCandidate(isNviCandidate(evaluatedCandidate),
+                                   toPendingCandidate(evaluatedCandidate));
+    }
```

Comment:
Do we need to pass boolean as variable to upsertCandidate method? I may be wrong, but I mean we can check if candidate is applicable, I mean this value is stored in DbCandidate object, so we can actually pass only DbCandidat to this method.

Reply:
Agreed

Reply:
Fixed